File: graph/router.py
from typing import Literal

from graph.state import ResearchState


def should_run_asp(state: ResearchState) -> Literal["reason_asp_conflicts", "synthesize"]:
    """Route to ASP conflict resolution if conflicts were detected, else synthesize directly."""
    if state.get("conflicts_detected"):
        return "reason_asp_conflicts"
    return "synthesize"

File: src/graph/router.ts
import type { ResearchStateType } from "./state.js";

/** Route to ASP conflict resolution if conflicts were detected, else skip to synthesize */
export function shouldRunAsp(
  state: ResearchStateType,
): "reason_asp_conflicts" | "synthesize" {
  return state.conflicts.length > 0 ? "reason_asp_conflicts" : "synthesize";
}
